src/HTMLElement/elements/HTMLAnchorElement.js

```diff
@@ -0,0 +1,160 @@
+import HTMLElement from '../../HTMLElement';
+import Url from '../../utils/Url';
+
+/**
+ * The HTMLAnchorElement interface represents hyperlink elements and provides special properties and methods
+ * (beyond those of the regular HTMLElement object interface they also have available to them by inheritance)
+ * for manipulating the layout and presentation of such elements.
+ *
+ * @see https://developer.mozilla.org/en/docs/Web/API/HTMLAnchorElement
+ * @class HTMLAnchorElement
+ */
+export default class HTMLAnchorElement extends HTMLElement {
+    constructor() {
+        super();
+        this._href = new Url();
+    }
+    /**
+     * @inheritDoc
+     */
+    setAttribute(key, value) {
+        super.setAttribute(key, value);
+        if (key === 'href') {
+            this.href = value;
+        }
+    }
+
+    /**
+     * Is a DOMString that reflects the href HTML attribute, containing a valid URL of a linked resource.
+     *
+     * @member {string} HTMLAnchorElement#href
+     * @returns {string}
+     */
+    get href() {
+        return this._href.toString();
+    }
+
+    /**
+     * @param {string} value
+     */
+    set href(value) {
+        this._href.href = value;
+        this._setAttribute('href', this._href.toString());
+        if (this.host === '' && this.ownerDocument) {
+            this._href.host = this.ownerDocument.location.host;
+        }
+
+        if (this.protocol === '' && this.ownerDocument) {
+            this._href.protocol = this.ownerDocument.location.protocol;
+        }
+    }
+
+    /**
+     * Is a DOMString representing the fragment identifier, including the leading hash mark ('#'), if any, in the
+     * referenced URL.
+     *
+     * @member {string} HTMLAnchorElement#hash
+    * @returns {string}
+     */
+    get hash() {
+        return this._href.hash;
+    }
+
+    /**
+     * @param {string} value
+     */
+    set hash(value) {
+        this._href.hash = value;
```

Comment:
Here is some more representative:

```javascript
var a=document.createElement('a');
undefined
a.href='/';
"/"
a.href
"https://github.com/"
a.hash='#some'
"#some"
a.href
"https://github.com/#some"
a.getAttribute('href')
"https://github.com/#some"
```

So as i understand a.getAttribute behaves different ways when a.hash or something changed. It's still relative when you do a.href='/something', but it changes to absoulte when you change it with .hash and other fields. What we gona do? Lets left current behaviour as is.